README.md

```diff
@@ -22,14 +22,79 @@ Tables will be automatically created.
 
 ## Getting started
 
-Install your database vendor specific plugin, you can use the Jenkins plugin site to search for it:
+To install the plugin login into Jenkins → Manage Jenkins → Manage Plugins → Available → Search for 'JUnit SQL Storage' → Install.
 
-https://plugins.jenkins.io/ui/search/?labels=database
+Use the following steps if you want to build and install the plugin from source.
+
+### Building
+
+To build the plugin use the `compile` goal
+```
+$ mvn clean compile
+```
+to run the tests use the `test` goal
+```
+$ mvn clean test
+```
+
+### Deploying Jenkins and the plugin
+
+To try out your chagnes you can deploy Jenkins using the `docker-compose` file provided in the repository.
+```
+$ docker compose up
+```
+This will deploy postgres, jaeger, and Jenkins servers. You'll need to monitor the output of the Jenkins container in 
+order to get the initial admin password.
 
-e.g. you could install the [PostgreSQL Database](https://plugins.jenkins.io/database-postgresql/) plugin.
+```
+$ docker logs -f junit-sql-storage-plugin_jenkins_1
+...
+*************************************************************
+Jenkins initial setup is required. An admin user has been created and a password generated.
+Please use the following password to proceed to installation:
+
+481093c2784447ddb058a99bdc50e878
+
+This may also be found at: /var/jenkins_home/secrets/initialAdminPassword
+*************************************************************
+```
+
+You can then open your browser to `http://localhost:8080` and enter the password to complete the setup (see next section).
+
+#### Setup Jenkins for the first time
+
+Once jenkins is deployed you can configure it using the `./setup_jenkins.sh` shell script provided in the repository.
+
+> **NOTE** you must first login to Jenkins and create an API token, you can do this by going to 
+> `http://localhost:8080/user/admin/configure` and clicking on 'Add new Token' under the 'API Token' section. Then you 
+> must copy the token and pass it as a parameter to the setup script.
+
+Here's an example of how you use the setup script:
+
+```
+./setup_jenkins.sh http://<hostname>:8080 <api_token>
```

Comment:
why not do this in the Dockerfile, 

Something like:
```
# install the plugin from update center to get all dependencies
RUN jenkins-plugin-cli --plugins junit-sql-storage configuration-as-code
# override with locally built version
COPY target/(whatever it is).hpi /usr/share/jenkins/ref/plugins/junit-sql-storage.jpi 

# configuration as code config for configuring this plugin
COPY docker/jenkins.yaml /usr/share/jenkins/ref/jenkins.yaml
```

Reply:
Wow thats great, I didn't know about the `jenkins-plugin-cli` binary or I would have done it this way :-/ I tried it out and it works fine.